src/components/fields/radio-field/radio-field.spec.js

```diff
@@ -0,0 +1,193 @@
+it('should render a radio field', () => {
+  const { getByLabelText } = renderRadioField();
+  expect(getByLabelText('RadioField')).toBeInTheDocument();
+});
+
+it('should render a title', () => {
+  const { getByText } = renderRadioField({ title: 'foo title' });
+  expect(getByText('foo title')).toBeInTheDocument();
+});
+
+it('should forward data-attributes', () => {
+  const { container } = renderRadioField({ 'data-foo': 'bar' });
+  expect(container.querySelector('[data-foo="bar"]')).toBeInTheDocument();
+});
+
+it('should have an HTML name', () => {
+  const { container } = renderRadioField({ name: 'foo' });
+  expect(container.querySelector('[name="foo"]')).toBeInTheDocument();
+});
+
+it('should call onFocus when the radio is focused', () => {
+  const onFocus = jest.fn();
+  const { getByLabelText } = renderRadioField({ onFocus });
+  getByLabelText(/RadioField Option 1/).focus();
+  expect(getByLabelText(/RadioField Option 1/)).toHaveFocus();
+  expect(onFocus).toHaveBeenCalled();
+});
+
+it('should call onBlur when radio loses focus', () => {
+  const onBlur = jest.fn();
+  const { getByLabelText } = renderRadioField({ onBlur });
+  getByLabelText(/RadioField Option 1/).focus();
+  expect(getByLabelText(/RadioField Option 1/)).toHaveFocus();
+  getByLabelText(/RadioField Option 1/).blur();
+  expect(getByLabelText(/RadioField Option 1/)).not.toHaveFocus();
+  expect(onBlur).toHaveBeenCalled();
+});
+
+it('should call onChange when changing the value', () => {
+  const { getByLabelText, onChange } = renderRadioField();
+  fireEvent.click(getByLabelText(/RadioField Option 2/));
+  expect(onChange).toHaveBeenCalled();
+});
+
+describe('when `description` is passed', () => {
+  it('should render a description', () => {
+    const { getByText } = renderRadioField({
+      description: 'foo description',
+    });
+    expect(getByText('foo description')).toBeInTheDocument();
+  });
+});
+
+describe('when `hint` is passed', () => {
+  it('should render a hint', () => {
+    const { getByText } = renderRadioField({ hint: 'foo hint' });
+    expect(getByText('foo hint')).toBeInTheDocument();
+  });
+});
+
+describe('when `badge` is passed', () => {
+  it('should render a badge', () => {
+    const { getByText } = renderRadioField({ badge: 'foo badge' });
+    expect(getByText('foo badge')).toBeInTheDocument();
+  });
+});
+
+describe('when disabled', () => {
+  it('should disable the input', () => {
+    const { getByLabelText } = renderRadioField({ isDisabled: true });
+    expect(getByLabelText(/RadioField Option 1/)).toHaveAttribute('disabled');
+    expect(getByLabelText(/RadioField Option 2/)).toHaveAttribute('disabled');
+  });
+});
+
+describe('when readOnly', () => {
+  it('should disable the input', () => {
+    const { getByLabelText } = renderRadioField({ isReadOnly: true });
+    expect(getByLabelText(/RadioField Option 1/)).toHaveAttribute('readonly');
+    expect(getByLabelText(/RadioField Option 2/)).toHaveAttribute('readonly');
```

Comment:
Are you sure we need this fuzzy matching everywhere? I understand the intend to have tests potentially more stable/flexible but I don't feel case for it really exists?

Reply:
I simply like it more like this 🙃 Do you see any problems with this approach?